test/explain/test_explainer_hetero.py

```diff
@@ -0,0 +1,204 @@
+import pytest
+import torch
+
+from torch_geometric.data import HeteroData
+from torch_geometric.explain import (
+    DummyExplainer,
+    Explainer,
+    HeteroExplanation,
+)
+from torch_geometric.explain.config import ExplanationType
+from torch_geometric.nn import SAGEConv, to_hetero
+
+
+def get_edge_index(num_src_nodes, num_dst_nodes, num_edges):
+    row = torch.randint(num_src_nodes, (num_edges, ), dtype=torch.long)
+    col = torch.randint(num_dst_nodes, (num_edges, ), dtype=torch.long)
+    return torch.stack([row, col], dim=0)
+
+
+@pytest.fixture
+def data():
+    data = HeteroData()
+    data['paper'].x = torch.randn(8, 16)
+    data['author'].x = torch.randn(10, 8)
+    data['paper', 'paper'].edge_index = get_edge_index(8, 8, 10)
+    data['author', 'paper'].edge_index = get_edge_index(10, 8, 10)
+    data['paper', 'author'].edge_index = get_edge_index(8, 10, 10)
+    return data
+
+
+class GraphSAGE(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = SAGEConv((-1, -1), 32)
+        self.conv2 = SAGEConv((-1, -1), 1)
+
+    def forward(self, x, edge_index):
+        x = self.conv1(x, edge_index).relu()
+        return self.conv2(x, edge_index)
+
+
+class HeteroSAGE(torch.nn.Module):
+    def __init__(self, metadata):
+        super().__init__()
+        self.graph_sage = to_hetero(GraphSAGE(), metadata, debug=False)
+
+    def forward(self, x_dict, edge_index_dict, *args) -> torch.Tensor:
+        return self.graph_sage(x_dict, edge_index_dict)['paper']
```

Comment:
Lets move this to a `conftest.py` and share across all tests (https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session). Happy to discuss more about this or even address it in another PR.

Reply:
Maybe best to move to a separate PR, I agree there is a lot of repeated code across the explanation tests that can be consolidated. Should I create an issue for this?

Reply:
Yeah feel free to add a TODO and also raise an issue.

Reply:
Created #6186